### DISCUSSION TITLE
DCM4CHEE doesnt transmit some Tags (such as TransducerData) in an C-Find Reqest

### DISCUSSION THREAD

**Discussion by weilbaum3000** [dcm4che/dcm4che#1395] [Q&A]
I tried to Query/Retrieve within a Python script using the pynetdicom library to execute serveral C-Find Requests. Unfortunatly the Response I received for the Requests contains all requested Tags but some of them are equals an empty string ("") eventhough the Paramters have values when i compare it with the DICOM-header in the WebViewer. Afterwards i executed a C-Find in the Ubuntu commandline (findscu from the library "dcmtk") to be sure and there occurs the same Problem. Even in wireshark this tag is completely missing.

**To Reproduce**
-Execute a C-Find on a dcm4chee PACS (AE: DCM4CHEE) and try to get the Parameter "TransducerData" (0018,5010) in the response. 

**Expected Behavior**
Normally this tag should contain one ore multiple string values in the response (normally as a list like **['string1' , 'string2' , 'string3']**), if it exists in the header of the image and is explicitly requested the executed C-Find. Probably dcm4chee doesnt event transmit this parameter because the wireshark doesnt show anything as well.

**System:**
 - OS: Ubuntu
 - Browser: Chrome
 - Python libraries: dcmtk, pynetdicom, pydicom

**Comment by gunterze** [comment]
If you want to get _(0018,5010)  Transducer Data_ already by DICOM Query Services (C-FIND or QIDO-RS), you have to add it to configured  **Attribute Tags** of the [Attribute Filter](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/attributeFilter.html) on Instance Level (**Attribute Entity**: `Instance`), so it get extracted from received objects into the DB instance record.

**Reply by weilbaum3000** [reply]
Hey thanks for your quick answer!

I think I already did as you said. Probably you got me wrong. The C-Find on Image/Instance Level already hast the Parameter TransducerData conigured in the dicom dataset in my python code. Here is a screenshot from the configuration:

![image](https://github.com/dcm4che/dcm4che/assets/69418567/ec04f996-e326-4238-8580-fdd00218b3a2)

All the other tags or at least most of them are transmitted right and occur in the response with the correct values. Concerning TransducerData it also occurs in the response but returns an empty value. To be sure I tried the same c-find request in the commandline wir finscu from the dcmtk-library and the same happend. The Value of the Tag was empty, but in the PACS DICOM header in the WebClient it isnt. Here is a screenshot of the request and the response in the ubuntu commandline:

findscu -S +sr -d --key Modality=US --key 0008,0052="IMAGE" --key 0020,000D=1.3.6.1.4.1.12559.11.4.1.1.446 --key 0020,000E=1.2.250.1.204.5.11510598007.20160405153658672017.1 --key 0008,0018=1.2.250.1.204.5.11510598007.20160405153914216098.0.9 --key 0018,5010 --call DCM4CHEE localhost 11112

To explain this request shortly: It extracts a specific image and I know this image has the tag transducerdata.
This is the response:

![image](https://github.com/dcm4che/dcm4che/assets/69418567/2613b5e8-fe39-424d-80bf-b8acb07e3175)

Also there the Value is empty.

Here is the wireshark protocol, where the tag had no value as well.

![Screenshot from 2024-01-09 15-13-38](https://github.com/dcm4che/dcm4che/assets/69418567/7920dca5-a7f6-4b40-871a-b066f6efd35a)

Best Regards
Philipp

**Reply by janossch** [reply]
I think you missunderstood the reply from @gunterze. He is suggesting to configure the **SCP**, in your case it is the dcm4chee archive server. In contrast you are configuring the **SCU** (your python code I believe), which is completly okay, but not enough. dcm4chee server doesn't return this attribute by default, so this is why you need to configure it. If you did that anyway, then well, I'm out of ideas. @gunterze correct me if I'm wrong here...

**Reply by weilbaum3000** [reply]
Ok thank you! Can you tell me how to configure that? 

Regards
Philipp

**Reply by janossch** [reply]
Sorry for the late reply. First of all I've not tried this, so you need to verify whether it is really working, but if you would like to configure I think you should open up the dcm4chee configuration UI first:

e.g.:  
https://localhost:8443/dcm4chee-arc/ui2

1. Menu -> Configuration -> 2. Devices tab -> 3. select archive server (dcm4chee-arc)
![image](https://github.com/dcm4che/dcm4che/assets/57434745/8f2e9a89-9ec1-4a92-98b6-ffa5417a81b1)

Extensions -> Device extension (Edit extension) -> Archive Device (Edit extension) -> Child objects -> Attribute Filter -> Select instance level attributes.
![image](https://github.com/dcm4che/dcm4che/assets/57434745/4eb3dc29-3986-463e-9fad-31ccf42518a3)

And there you should see a lot of instance level attributes where you can add your desired one (TransducerData) and finally save it.

Hope this helps!
